pkg/sql/tenant_creation.go

```diff
@@ -601,6 +601,12 @@ HAVING ($1 = '' OR NOT EXISTS (SELECT 1 FROM system.tenants t WHERE t.name = $1)
 	if uint64(lastIDFromSequence+1) > nextIDFromTable {
 		nextID = uint64(lastIDFromSequence + 1)
 	}
+	// ID 2 is reserved for future use: it was a "template" tenant in 23.2 UA and
+	// used for an internal test in serverless, so we can reclaim it if we want it
+	// so long as we don't allow it to be used for real tenants.
+	if nextID == 2 {
```

Comment:
Can you add a test for this? E.g. ensure the first created tenant has the id 3.

Reply:
we have lots of tests that will assert this in our logic tests which is why this is so red right now, since I still need to update those; I was just wanting to get directional approval before I went and ran `--rewrite` for 45 minutes.